errors.go

```diff
@@ -370,19 +371,6 @@ func getTopLevelPrinter(err error) func(*bytes.Buffer) bool {
 	}
 }
 
-func getCause(e error) error {
-	if uw, ok := e.(unwrapper); ok {
-		return uw.Unwrap()
-	}
-
-	var wrapped Error
-	if errors.As(e, &wrapped) {
-		return wrapped
-	}
```

Comment:
`getCause` was formerly used to pull a wrapped error out of an `error` passed to the API.  We then used `wrappingError` to ensure we were able to implement our own `Error` interfacing while maintaining existing wrapping relationships (in part so that `errors.Is` and `errors.As` would still function as expected).

The block of code I've highlighted here may have resulted in an error wrapping itself (because `getCause` may return true if `e` implements `Error`).

I'm not sure how much sense that all made... there are a lot of different definitions of "error" used above.  I'm happy to take another crack at explaining that.  The TL;DR is that this block of code means `getCause` may not always return the cause.  The correct thing to do (after the changes in 1cdac23a47183743fb4c0074c2aeaee78b744f8c) is just try to unwrap the error.  This reduces `getCause` enough that I felt like it made sense to just get rid of it.

Reply:
Probably worth mentioning that the case I talked about above could actually not occur in practice given the current state of the code.  There is a check before `getCause` which happens to ensure that `getCause` could not be passed an `Error`.  But that's happenstance and not something `getCause` is supposed to rely on.

But yeah, this `errors.As` call just doesn't really return the cause of `e`, so we should get rid of it.